src/ip/QosConfig.cc

```diff
@@ -144,6 +201,18 @@ Ip::Qos::getNfMarkCallback(enum nf_conntrack_msg_type,
 
     return NFCT_CB_CONTINUE;
 }
+
+int
+Ip::Qos::getNfClientConnmarkCallback(enum nf_conntrack_msg_type,
+                           struct nf_conntrack *ct,
+                           void *client_connmark)
+{
+    nfmark_t *mark = (nfmark_t *)client_connmark;
```

Comment:
Please use appropriate C++ casting (static_cast here) in new and updated code instead of C casts.